### PR TITLE
Getter/setter for user weights

### PR DESCRIPTION
**Pull request by klendathu2k**
Adding a "get" and "set" method for the user weights defined in the event generator record.  These were previously missing, and will be required for upcoming hard diffraction simulations.

### CODE REVIEW

File: StRoot/StarGenerator/EVENT/StarGenEvent.h
Comment (plexoos) [inline StRoot/StarGenerator/EVENT/StarGenEvent.h]:
```suggestion
  const std::vector<double>& GetUserWeights() const { return mWeights; }
```
If you want to be a c++ purist 🐕 `

**Review by plexoos — APPROVED** [review]

**Comment by plexoos** [comment]
I guess waiting for Victor's approval on this unambiguous addition is futile...

File: StRoot/StarGenerator/EVENT/StarGenEvent.h
Comment (plexoos) [inline StRoot/StarGenerator/EVENT/StarGenEvent.h:212]: Ah shoot, just noticed after merging. This getter signature does not make much sense and should be removed or rewritten to return a non-const reference

**Comment by perevbnlgov** [comment]
Well, if such setter/getter is needed, no objection. The only remmark, it could be slightly safer to make the getter const

**Pull request merged**
